src/averageAges.js

```diff
@@ -15,11 +15,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(({ sex, died }) => (sex === 'm' && (
+    century
+      ? century === Math.ceil(died / 100)
+      : true
+  )));
+
+  const menAges = men.map(({ born, died }) => died - born);
```

Comment:
Let's avoid unnecessary array iterating. You need this array of ages only to calculate average age using reduce method, so why don't calculate age in reduce method? 🙂